app/controllers/application_controller.rb

```diff
@@ -447,7 +447,7 @@ def report_only
     end
     # Dashboard widget will send in report result id else, find report result in the sandbox
     search_id = params[:rr_id] ? params[:rr_id].to_i : @sb[:pages][:rr_id]
-    rr = MiqReportResult.for_user(current_user).find(search_id)
```

Comment:
This looks like you are giving access to any report result to any user. All he has to do is to provide the `?rr_id=` value